### PR TITLE
feat: allow sourceMaps with sourceRoot (#31)

### PR DESCRIPTION
**Pull request by Toxicable**

### CODE REVIEW

**Comment by Toxicable** [comment]
I wasn't able to run the testing suite locally, so pushing it to CI.

Not sure how else to specifically assert it resolved the sourceRoot corrrectly, apaart from assueming it'll throw if it fails to resolve it

**Comment by Toxicable** [comment]
ok so getting this failure in CI aswell
Looks like some file has some random bytes at the top of it?
I commented out my test and still getting it, but reset back to upstream/master and it fixed it.

```
test/v8-to-istanbul.js 2> undefined:1
test/v8-to-istanbul.js 2> n����q�
test/v8-to-istanbul.js 2>  ^
test/v8-to-istanbul.js 2> 
test/v8-to-istanbul.js 2> SyntaxError: Unexpected token � in JSON at position 1
test/v8-to-istanbul.js 2>     at JSON.parse (<anonymous>)
test/v8-to-istanbul.js 2>     at Object.parseSourceMapInput (/Users/fabianwiles/Documents/v8-to-istanbul/node_modules/source-map-support/node_modules/source-map/lib/util.js:433:15)
test/v8-to-istanbul.js 2>     at new SourceMapConsumer (/Users/fabianwiles/Documents/v8-to-istanbul/node_modules/source-map-support/node_modules/source-map/lib/source-map-consumer.js:17:22)
test/v8-to-istanbul.js 2>     at mapSourcePosition (/Users/fabianwiles/Documents/v8-to-istanbul/node_modules/source-map-support/source-map-support.js:190:14)
test/v8-to-istanbul.js 2>     at wrapCallSite (/Users/fabianwiles/Documents/v8-to-istanbul/node_modules/source-map-support/source-map-support.js:358:20)
test/v8-to-istanbul.js 2>     at /Users/fabianwiles/Documents/v8-to-istanbul/node_modules/source-map-support/source-map-support.js:399:26
test/v8-to-istanbul.js 2>     at Array.map (<anonymous>)
test/v8-to-istanbul.js 2>     at Function.prepareStackTrace (/Users/fabianwiles/Documents/v8-to-istanbul/node_modules/source-map-support/source-map-support.js:398:30)
test/v8-to-istanbul.js 2>     at process.emit (/Users/fabianwiles/Documents/v8-to-istanbul/node_modules/source-map-support/source-map-support.js:457:52)
test/v8-to-istanbul.js 2>     at processEmit [as emit] (/Users/fabianwiles/Documents/v8-to-istanbul/node_modules/signal-exit/index.js:155:32)
test/v8-to-istanbul.js 2>     at process._fatalException (internal/bootstrap/node.js:493:27)
```

File: test/v8-to-istanbul.js [hunk 1]
```diff
@@ -28,6 +30,28 @@ describe('V8ToIstanbul', () => {
...
+const a=(cov_1myqaytex8.s[0]++,(cov_1myqaytex8.b[0][0]++,99)||(cov_1myqaytex8.b[0][1]++,33));// basic conditional operation.
```
Comment (bcoe) [inline test/v8-to-istanbul.js]: I haven't done any digging yet, but I bet you could make a more contrived example where you just have a file with:

```js
const foo.js
// sourceMap=data=abc123
```

☝️ I'd then just put this in the fixtures directory, and have it so it remaps to another file in the fixtures directory.

Comment (Toxicable) [reply test/v8-to-istanbul.js]: fair point about a smaller file
however, it still has to be generated since the sourceRoot is an absolute path

**Comment by Toxicable** [comment]
So looks like the error I was having with `SyntaxError: Unexpected token � in JSON at position 1` was because `source-map-support` was trying to parse the string I was writing as a source map as an actual source map fir that file.

File: test/v8-to-istanbul.js [hunk 2]
```diff
@@ -28,6 +31,33 @@ describe('V8ToIstanbul', () => {
...
+      const absoluteSourceFilePath = path.join(sourceRoot, sourceFileName)
+      const map = new sourcemap.SourceMapGenerator({
+        file: sourceFileName,
+        sourceRoot
+      })
+      map.addMapping({
+        source: sourceFileName,
+        original: { line: 1, column: 1 },
+        generated: { line: 1, column: 1 }
+      })
```
Comment (bcoe) [inline test/v8-to-istanbul.js]: I'd be tempted to just pre-generate this data, and write it to a fixture ... but, at the same time, I like that this is a useful demonstration of how to auto-generate this information.

Not a blocker for me.

**Pull request merged**